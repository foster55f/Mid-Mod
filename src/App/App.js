import React, { Component } from 'react';
import './App.css';
import ReservationContainer from '../ReservationContainer/ReservationContainer';
import ReservationForm from '../ReservationForm/ReservationForm';

class App extends Component {
  constructor() {
    super();
    this.state = {
      data: null
    }
  }

  componentDidMount() {
    fetch('http://localhost:3001/api/v1/reservations')
      .then(response => response.json())
      .then(data => this.setState({ data }))
      .catch(error => console.log(error))
  }

  addReservation = (newReservation) => {
    console.log(newReservation.id)
    this.setState({ data: [...this.state.data, newReservation] });
    fetch('http://localhost:3001/api/v1/reservations', {
    method: 'POST',
    headers: {
      'Accept': 'application/json',
      'Content-Type': 'application/json'
    },
      body: JSON.stringify({
        "id": newReservation.id,
        "name": newReservation.name,
        "date": newReservation.date,
        "time": newReservation.time,
        "number": parseInt(newReservation.number)
      })
    }).then(res => res.json())
    .then(data =>  data)
    .catch(error =>error)
  }

  

  render() {
    return (
      <div className="App">
        <h1 className='app-title'>Turing Cafe Reservations</h1>
        {this.state.data && (<ReservationContainer reservations={this.state.data} />)}
        <div className='resy-form'>
          <ReservationForm addReservation={this.addReservation}/>
        </div>
        <div className='resy-container'>      
        </div>
      </div>
    )
  }
}

export default App;
