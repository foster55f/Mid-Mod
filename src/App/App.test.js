import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  
  it('should match the snapshot', () => {
    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should update the state with a new reservation when addReservation is called', () => {
    const wrapper = shallow(<App />)
    const mockReservation = { id: 55998882, name: 'foster', date: "2019-12-15", time: "5:00", number: "55" };
    const expected=[mockReservation]
    expect(wrapper.state('reservations')).toEqual();
    wrapper.instance().addReservation(mockReservation)
    expect(wrapper.state('reservations')).toEqual([expected]);
  })
})
