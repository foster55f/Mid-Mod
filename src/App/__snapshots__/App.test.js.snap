// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <h1
    className="app-title"
  >
    Turing Cafe Reservations
  </h1>
  <div
    className="resy-form"
  >
    <ReservationForm
      addReservation={[Function]}
    />
  </div>
  <div
    className="resy-container"
  />
</div>
`;

exports[`App should match the snapshot 2`] = `
<div
  className="App"
>
  <h1
    className="app-title"
  >
    Turing Cafe Reservations
  </h1>
  <div
    className="resy-form"
  >
    <ReservationForm
      addReservation={[Function]}
    />
  </div>
  <div
    className="resy-container"
  />
</div>
`;

exports[`App should update the state 1`] = `
<div
  className="App"
>
  <h1
    className="app-title"
  >
    Turing Cafe Reservations
  </h1>
  <div
    className="resy-form"
  >
    <ReservationForm
      addReservation={[Function]}
    />
  </div>
  <div
    className="resy-container"
  />
</div>
`;

exports[`App should update the state with a new reservation 1`] = `
<div
  className="App"
>
  <h1
    className="app-title"
  >
    Turing Cafe Reservations
  </h1>
  <div
    className="resy-form"
  >
    <ReservationForm
      addReservation={[Function]}
    />
  </div>
  <div
    className="resy-container"
  />
</div>
`;

exports[`App should update the state with a new reservation when addReservation is called 1`] = `
<div
  className="App"
>
  <h1
    className="app-title"
  >
    Turing Cafe Reservations
  </h1>
  <div
    className="resy-form"
  >
    <ReservationForm
      addReservation={[Function]}
    />
  </div>
  <div
    className="resy-container"
  />
</div>
`;

exports[`App should update the state with a newR 1`] = `
<div
  className="App"
>
  <h1
    className="app-title"
  >
    Turing Cafe Reservations
  </h1>
  <div
    className="resy-form"
  >
    <ReservationForm
      addReservation={[Function]}
    />
  </div>
  <div
    className="resy-container"
  />
</div>
`;
